atlas/templates/Makefile.common.gotmpl

```diff
@@ -7,8 +7,9 @@ DOCKERFILE_PATH         ?= $(CURDIR)/docker
 
 # configuration for image names
 USERNAME                ?= $(USER)
+BUILD_NUMBER            ?= 0
 GIT_COMMIT              ?= $(shell git describe --dirty=-unsupported --always --tags || echo pre-commit)
-IMAGE_VERSION           ?= $(GIT_COMMIT)-$(USERNAME)
+IMAGE_VERSION           ?= $(GIT_COMMIT)-j$(BUILD_NUMBER)
```

Comment:
Images are immutable, let's leave off build number. Developers will also find out that re-pushing the same image tag will result in errors retrieving the image. They can get cached version in a docker registry mirror and k8s. 
If developer wants to push a new image, they need to use git. We should be erroring on push events for images with `-unsupported` in the name.
```
make push // Image is in a registry
// make some changes
make push // error
git commit -am "commit my changes"
make push // success
```